src/main/java/courseManager/InstanceList.java

```diff
@@ -0,0 +1,143 @@
+
+    public InstanceList(String pluralTitle, int totalNumberOfInstances) {
+        this.listOfAssessmentInstances = new ArrayList<>();
+        this.singularTitle = toSingular(pluralTitle);
+        for (int i = 0, j = 1; i < totalNumberOfInstances; i++, j++){
+            this.addInstance(this.singularTitle + " #" + j);
+        }
+    }
+
+    public String toSingular(String pluralTitle){
+        if (pluralTitle == "Quizzes"){
+            return "Quiz";
+        }
+        if (pluralTitle.charAt(pluralTitle.length()-1) == 's'){
+            return singularTitle.substring(0, pluralTitle.length()-1);
+        }
+        return singularTitle;
+    }
+
+    public int getTotalNumberOfInstances() {
+        return totalNumberOfInstances;
+    }
+
+    public int getNumberOfCommittedInstances() {
+        int numberOfCommittedInstances = 0;
+        for (AssessmentInstance instance : listOfAssessmentInstances) {
+            if (instance.isCommitted()) {
+                numberOfCommittedInstances++;
+            }
+        }
+        return numberOfCommittedInstances;
+    }
+
+    public int getNumberOfSubmittedInstances() {
+        int numberOfSubmittedInstances = 0;
+        for (AssessmentInstance instance : listOfAssessmentInstances) {
+            if (instance.isSubmitted()) {
+                numberOfSubmittedInstances++;
+            }
+        }
+        return numberOfSubmittedInstances;
+    }
+
+    public double[] getAllMarks() {
+        double[] allMarks = new double[totalNumberOfInstances];
+        for (int i = 0; i < totalNumberOfInstances; i++) {
+            allMarks[i] = listOfAssessmentInstances.get(i).getMark();
+        }
+        return allMarks;
+    }
+
+
+    public double[] getCommittedMarks() {
+        double[] committedMarks = new double[getNumberOfCommittedInstances()];
+        int i = 0;
+        for (AssessmentInstance instance : listOfAssessmentInstances) {
+            if (instance.isCommitted()) {
+                committedMarks[i] = instance.getMark();
+                i++;
+            }
+        }
+        return committedMarks;
+    }
+
+    public void addInstance(String name) {
+        AssessmentInstance instance = new AssessmentInstance(name);
+        listOfAssessmentInstances.add(instance);
+        totalNumberOfInstances++;
+    }
+
+    public void removeInstance(int index) {
+        if (listOfAssessmentInstances.get(index).isCommitted()) {
+            numberOfCommittedInstances--;
+        }
+        if (listOfAssessmentInstances.get(index).isSubmitted()) {
+            numberOfSubmittedInstances--;
+        }
+        listOfAssessmentInstances.remove(index);
+        totalNumberOfInstances--;
+    }
+
+    public AssessmentInstance getInstanceData(int index) {
+        return listOfAssessmentInstances.get(index);
+    }
+
+    /*overloaded methods to edit particular data points*/
+    /*if there is a more elegant solution to this, feel free to change it! I really don't like this lol*/
```

Comment:
A (very slightly) more elegant solution is to have separate editInstanceName, editInstanceDate and editInstanceMark functions. although we'd also need getters for those since the GUI needs to display all that information. Honestly since the AssessmentInstances are already pretty nicely encapsulated it's not the end of the world if we just directly return the instances themselves. Yes, not great on coupling but really good on readability and ease of API.

Reply:
Done! I also think we could just have specific functions for each variable, especially since I now added Alan's suggestion of having a LocalTime variable as well